File: FizzBuzz/C#/FizzBuzz.cs
using System;
using System.Linq;

class Program
{
	public static void FizzBuzz(int N)
	{
		foreach (int i in Enumerable.Range(1, N))
		{
			if(i % 3 == 0 && i % 5 ==0)
				Console.WriteLine("FizzBuzz");
			else if(i % 3 == 0)
				Console.WriteLine("Fizz");
			else if(i % 5 == 0)
				Console.WriteLine("Buzz");
			else
				Console.WriteLine(i);
		}
	}

	public static void Main(string[] args)
	{
		FizzBuzz(100);
	}
}

File: FizzBuzz/Haskell/FizzBuzz.hs
-- Based on a lecture: https://www.youtube.com/watch?v=1jZ7j21g028
fizzes = cycle [ "", "", "Fizz" ]
buzzes = cycle [ "", "", "", "", "Buzz" ]
pattern = zipWith (++) fizzes buzzes
combine word number = if null word then show number else word

fizzBuzz :: Int -> [String]
fizzBuzz n = zipWith combine pattern [1..n] 

main :: IO()
main = print(fizzBuzz 100)
